feat: Add edit and delete post feature

Naman-Saxena1 opened the pull request:
# Added Frontend and Backend for edit and delete post
1. Logged in users can edit their posts
2. Logged in users can delete their posts

## Files to review
1. EditPostModal.jsx
2. UserPost.jsx

## Login creds to test
Feel free to test functionalities using creds
### User 1 (Alexa)
Email - newuser1@gmail.com
Password - zxcv

### User 2 (Jane)
Email - newuser2@gmail.com
Password - zxcv

## Screenshots
![Capture5](https://user-images.githubusercontent.com/90456667/165105095-104d634e-7ae6-4b37-ab87-f792f2ce3b1b.PNG)
![Capture6](https://user-images.githubusercontent.com/90456667/165105110-201d340e-31c0-4df4-9710-bf063cef9f7e.PNG)

Naman-Saxena1 committed:
feat: Add edit post feature

Naman-Saxena1 committed:
feat: Delete post feature

bhtibrewal commented on src/components/EditPostModal/EditPostModal.css:
```diff
@@ -0,0 +1,117 @@
+    top: 0;
+    width: 100%;
+    height: 100%;
+    z-index: var(--zindex-edit-post-modal);
```
Appreciation: it's nice you have used variable for z-index

bhtibrewal commented on src/components/EditPostModal/EditPostModal.jsx:
```diff
@@ -0,0 +1,115 @@
+    } = useEditModal()
+
+    const [updatedPostText, setUpdatedPostText ] = useState(contentText)
+    const dispatch = useDispatch()
```
Suggestion: it would better if you name it more specifically

bhtibrewal commented on src/components/UserPost/UserPost.css:
```diff
@@ -62,12 +67,57 @@
+    flex-direction: column;
+    justify-content: flex-start;
+    background-color: var(--white);
+    box-shadow: 0 0 1px black;
```
Suggestion: use var for color for consistency, also you shouldn't use complete black or white anywhere

Naman-Saxena1 replied:
Done

bhtibrewal commented on src/components/UserPost/UserPost.css:
```diff
@@ -62,12 +67,57 @@
+{
+    margin: 0;
+    flex-grow: 1;
+    font-size: 12px;
```
shouldn't you be using em here for font-size, because if in case the root font size changes this will stay at 12px

Naman-Saxena1 replied:
Done

bhtibrewal commented on src/components/UserPost/UserPost.jsx:
```diff
@@ -119,6 +140,20 @@ function UserPost({userPostDetails})
         }
     }
 
+    const deleteUserPost = async() => {
```
Suggestion: you should move these functions to a seperate file

bhtibrewal reviewed (changes requested):
Code is good, changes required on the UI 
UI suggestions: 
- I don't find the use of home icon in the header when home is already there in sidebar, 
- I don't think it would good to give logout button like a CTA button  in the header, instead it would be better if you give it in the sidebar.
- It doesn't makes sense in the first place to give user the option to edit/delete when he can't as it's not his post. Instead there should be other options like bookmark post ,etc.
- it takes up a lot of place if you show all the comments on the post there itself, you can show all on a single post page 

Great work on the comments and reply but there is one issue that the input field doesn't clear the instant user clicks enter, instead after the comment is posted so i ended up clicking enter multiple times, so fix this, also there is no button to post the comment along side the input field so how will it work on mobile

yaswanthmyneni commented on src/components/EditPostModal/EditPostModal.css:
```diff
@@ -0,0 +1,117 @@
+    max-height: 28rem;
+    display: flex;
+    flex-direction: column;
+    padding: 0 0 0 2rem;
```
suggestion: can give just padding-left here

Naman-Saxena1 replied:
Done

yaswanthmyneni reviewed (approved):
the code is good, you can merge.

Naman-Saxena1 committed:
refactor: Refactor as per PR reviews

Naman-Saxena1 commented:
> Code is good, changes required on the UI UI suggestions:
> 
> * I don't find the use of home icon in the header when home is already there in sidebar,
> * I don't think it would good to give logout button like a CTA button  in the header, instead it would be better if you give it in the sidebar.
> * It doesn't makes sense in the first place to give user the option to edit/delete when he can't as it's not his post. Instead there should be other options like bookmark post ,etc.
> * it takes up a lot of place if you show all the comments on the post there itself, you can show all on a single post page
> 
> Great work on the comments and reply but there is one issue that the input field doesn't clear the instant user clicks enter, instead after the comment is posted so i ended up clicking enter multiple times, so fix this, also there is no button to post the comment along side the input field so how will it work on mobile

1. Home button was for Login and Signup page where sidebar is not present. Thanks for pointing out, removed it from other pages.
2. Sure, will change that when I create user profile feature and profile page
3. Thanks for pointing out, removed edit/delete feature from posts that are from other users
4. Yeah, I will do that when I make single post page like twitter implementation, this is temporary
5. Thanks for pointing out issue with comments and replies, refactored code and added "Send" button to add comment/reply rather than enter key press.

Thanks for giving detailed review 😄

Naman-Saxena1 committed:
refactor: Refactor code as PR review
